### PR TITLE
Add fiat onramp

### PR DESCRIPTION
**Pull request by Bitcoinera**
Still having some problems since the widget is constantly reloading. But this might be because of the localhost environment.
Anyway, I am not even sure we are going to use this integration, since they are asking for a KYC to be able to buy crypto :shrug: 

image

### CODE REVIEW

File: config/env.js
Comment (creed-victor) [inline config/env.js:79]: As this env variable is used in browser side you can prefix variable with REACT_APP_ ex: `process.env.REACT_APP_TRANSAK_API_KEY`.
This way there is no need to add custom environment keys to config files, can freely use .env files.

Variables without prefix is considered secure, will be used only by node and will not be passed to browser.

File: src/app/containers/BuyBTCPage/index.tsx
Comment (creed-victor) [inline src/app/containers/BuyBTCPage/index.tsx:14]: Having this in `components` folder is a little bit deceitful, file content is more like provider, service or util in my eyes

Comment (creed-victor) [reply src/app/containers/BuyBTCPage/index.tsx:14]: ok, maybe it's a component but then it's content is wrong - will leave comment there.

**Comment by Bitcoinera** [comment]
Wow, very insightful review! Thank you so much, @creed-victor !

File: src/app/containers/BuyBTCPage/components/transak.ts
Comment (Bitcoinera) [reply src/app/containers/BuyBTCPage/components/transak.ts:32]: I get an error when writing `  return <></>;` Like it is not valid JS (it is HTML actually). Why don't you get the same error? :thinking:

**Comment by Bitcoinera** [comment]
> Wrap initialisation in useEffect hook, this should prevent widget to reload.

Unfortunately this didn't make the trick :disappointed:

File: src/app/containers/BuyBTCPage/components/transak.ts
Comment (grinry) [reply src/app/containers/BuyBTCPage/components/transak.ts:32]: File extension then should be .tsx and function name start with uppercase - OpenTransak instead of openTransak.

To use this component use <OpenTransak /> instead of {openTransak()}.

Most likely useEffect didn't do its job because of this

**Comment by Bitcoinera** [comment]
It is working now! :tada: 

Guess will need to buy something to try 100%

File: src/app/containers/BuyBTCPage/components/transak.ts
Comment (Bitcoinera) [reply src/app/containers/BuyBTCPage/components/transak.ts]: well, I guess then I need to put it back in the components folder

**Comment by Bitcoinera** [comment]
Ah no, it seems it was all a mirage. I'm getting this because of rendering too many times the component (note line 39, where I'm doing `OpenTransak()`, calling directly the code). It seems it doesn't like it

File: src/app/containers/BuyBTCPage/index.tsx
Comment (creed-victor) [inline src/app/containers/BuyBTCPage/index.tsx:39]: It doesn't refresh to me anymore (preview: https://sovryn-dapp-gblskn032-sovryn.vercel.app/).
But maybe this can help:
```suggestion
            <div className="col-12 mt-2"><OpenTransak /></div>
```

File: src/app/containers/BuyBTCPage/index.tsx
Comment (creed-victor) [inline src/app/containers/BuyBTCPage/index.tsx:45]: Actually, it can be simplified like this:

```suggestion
          <div className="row">
            <div className="col-12 mt-2">{connected && <OpenTransak />}</div>
          </div>
        )
```

File: src/app/containers/BuyBTCPage/index.tsx
Comment (grinry) [inline src/app/containers/BuyBTCPage/index.tsx:39]:
```suggestion
```

File: src/react-app-env.d.ts
Comment (grinry) [inline src/react-app-env.d.ts:9]:
```suggestion
```

**Review by grinry — APPROVED** [review]

**Comment by Bitcoinera** [comment]
I had to change the default crypto and the environment parameter to an env variable

**Comment by grinry** [comment]
> I had to change the default crypto and the environment parameter to an env variable.

Include new env variable to .env files too (at least to .env.example) :)

> Some errors sometimes with Intercom (from Transak), but this must be because of the localhost environment
> 

It may also happen if you have adblocker installed

**Comment by Bitcoinera** [comment]
> It may also happen if you have adblocker installed

Ah yeah, makes sense. I use Brave. OK, so all good :+1:

File: src/locales/pt_br/translation.json
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:723]:
```suggestion
    "title": "Receber rBTC",
```

File: src/locales/pt_br/translation.json
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:724]:
```suggestion
    "subtitle": "Deposite BTC ou dinheiro fiduciário para receber rBTC na sua carteira",
```

File: src/locales/pt_br/translation.json
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:739]:
```suggestion
    "buttonBTC": "Enviar BTC para receber rBTC",
```

File: src/locales/pt_br/translation.json
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:740]:
```suggestion
    "buttonFiat": "Comprar rBTC usando dinheiro",
```

**Review by bitcoinuser — CHANGES_REQUESTED** [review]
Changes PT-BR

File: src/locales/pt_br/translation.json
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:732]:
```suggestion
    "subtitle": "Receba rBTC na sua carteira depositando BTC ou realizando uma compra com dinheiro",
```

File: src/locales/pt_br/translation.json
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:751]:
```suggestion
    "buttonBTC": "Enviar BTC",
```

File: src/locales/pt_br/translation.json
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:752]:
```suggestion
    "buttonFiat": "Comprar rBTC",
```

File: src/locales/pt_br/translation.json
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:745]:
```suggestion
      "title": "Comprar rBTC:",
```

File: src/locales/pt_br/translation.json
Comment (bitcoinuser) [inline src/locales/pt_br/translation.json:746]:
```suggestion
      "explanation": "Ao clicar no botão abaixo, você será direcionado para um serviço de compra de BTC. Certifique-se de que o endereço BTC para o qual você está enviando seus fundos seja o mesmo que o abaixo. A Sovryn fará a conversão para rBTC para você. Fique atento ao limite de 0,01"
```

**Review by bitcoinuser — CHANGES_REQUESTED** [review]
Changes PT-BR

**Comment by bitcoinuser** [comment]
Hi, @creed-victor please approve my translations here. Thanks

File: .env.example
Comment (creed-victor) [inline .env.example:3]:
```suggestion
REACT_APP_TRANSAK_ENV=PRODUCTION

```

**Comment by creed-victor** [comment]
Converted to draft until we will have api keys ready to not merge it by mistake.

**Comment by Bitcoinera** [comment]
Yuhuuu :rocket:

**Pull request merged**
